### PR TITLE
RM-7882: Update unit tests to make the ArgumentException message assertions compatible then .NET 5

### PR DESCRIPTION
**Pull request by derlistigelurch**

### CODE REVIEW

File: Remotion/Data/DomainObjects.UnitTests/EventReceiver/SequenceEventReceiver.cs [hunk 1]
```diff
@@ -120,7 +120,7 @@ public void Check (ChangeState[] expectedStates)
         {
           Assert.Fail (
               string.Format (
-                  "{0}\r\nExpected state: {1} - {2} - {3} \r\nActual state: {4} - {5} - {6}",
+                  "{0}\r\nExpected state: {1} - {2} - {3} ", "state: {4} - {5} - {6}",
```
Comment (a-ctor) [inline Remotion/Data/DomainObjects.UnitTests/EventReceiver/SequenceEventReceiver.cs:123]: The replace caught something fishy :D

File: Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/IntegrationTests/SqlProviderExecuteCollectionQueryTest.cs [hunk 2]
```diff
@@ -74,7 +75,7 @@ public void AllDataTypes ()
       query.Parameters.Add ("@int64", 9223372036854775807L);
       query.Parameters.Add ("@singleLowerBound", (float) 6789);
       query.Parameters.Add ("@singleUpperBound", (float) 6790);
-      query.Parameters.Add ("@string", "abcdef���");
+      query.Parameters.Add ("@string", "abcdeföäü");
```
Comment (a-ctor) [inline Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/IntegrationTests/SqlProviderExecuteCollectionQueryTest.cs:78]: There seems to be some encoding change here, we should revert the change.

File: Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/IntegrationTests/SqlProviderGeneratedSqlTestHelper.cs [hunk 3]
```diff
@@ -105,7 +105,7 @@ public void CheckCommand (IDbCommand sqlCommand, string expectedSql, params Tupl
         Assert.That (
             sqlCommand.CommandText,
             Is.EqualTo (expectedSql),
-            "Command text doesn't match.\r\nActual statement: {0}\r\nExpected statement: {1})",
+            "Command text doesn't match.", "statement: {0}\r\nExpected statement: {1})",
```
Comment (a-ctor) [inline Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/IntegrationTests/SqlProviderGeneratedSqlTestHelper.cs:108]: ❌🐟

File: SharedSource/UnitTests/Development/NUnit/NUnitExtensionTests.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+namespace Remotion.UnitTests.Development.NUnit
+{
+  [TestFixture]
+  public class NUnitExtensionTests
+  {
+    [Test]
+    public void AssertThatMessageContainsWhitespaces_MessageWithWhitespaces ()
+    {
+      var message = "message with whitespaces";
+      Assert.DoesNotThrow (() => NUnitExtensions.AssertThatMessageContainsWhitespaces (message));
```
Comment (a-ctor) [inline SharedSource/UnitTests/Development/NUnit/NUnitExtensionTests.cs:30]: I can't find another usage of this function in re-motion. I don't think we need to wrap code that should not throw in an assert as the test will fail anyway when an exception is uncaught.

File: SharedSource/UnitTests/Development/NUnit/NUnitExtensionTests.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    {
+      var message = "messageWithoutWhitespaces";
+      Assert.Throws<InvalidOperationException> (() => NUnitExtensions.AssertThatMessageContainsWhitespaces (message));
+    }
+
+    [Test]
+    public void AssertThatParameterDoesNotContainWhitespaces_ParameterWithoutWhitespaces ()
+    {
+      var paramName = "paramName";
+      Assert.DoesNotThrow (() => NUnitExtensions.AssertThatParameterDoesNotContainWhitespaces (paramName));
```
Comment (a-ctor) [inline SharedSource/UnitTests/Development/NUnit/NUnitExtensionTests.cs:44]: See above

**Review by a-ctor — CHANGES_REQUESTED** [review]

File: Remotion/Core/Reflection.CodeGeneration.UnitTests/ClassEmitterTest.cs [hunk 6]
```diff
@@ -20,6 +20,7 @@
 using Castle.DynamicProxy.Generators.Emitters;
 using Castle.DynamicProxy.Generators.Emitters.SimpleAST;
 using NUnit.Framework;
+using Remotion.Development.UnitTesting.NUnit;
```
Comment (a-ctor) [inline Remotion/Core/Reflection.CodeGeneration.UnitTests/ClassEmitterTest.cs:23]: There are a lot of leftover changes due to the revert where this using import is the only change in the file.

**Review by a-ctor — CHANGES_REQUESTED** [review]

**Review by a-ctor — APPROVED** [review]

File: Remotion/ObjectBinding/UnitTests/BusinessObjectProviderAttributeTest.cs [hunk 7]
```diff
@@ -45,9 +46,9 @@ public void Initialize_WithInvalidType ()
       Assert.That (
           () => new StubBusinessObjectProviderAttribute (typeof (object)),
           Throws.ArgumentException
-              .With.Message.EqualTo (
-                  "Parameter 'businessObjectProviderType' is a 'System.Object', which cannot be assigned to type 'Remotion.ObjectBinding.IBusinessObjectProvider'."
-                  + "\r\nParameter name: businessObjectProviderType"));
+              .With.ArgumentExceptionMessageEqualTo (
+                  "Parameter 'businessObjectProviderType' is a 'System.Object', which cannot be assigned to type 'Remotion.ObjectBinding.IBusinessObjectProvider'.",
+                 "businessObjectProviderType"));
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/UnitTests/BusinessObjectProviderAttributeTest.cs:51]: whitespace/indent

File: Remotion/ObjectBinding/UnitTests/BusinessObjectProviderTest.cs [hunk 8]
```diff
@@ -153,9 +154,9 @@ public void SetProvider_WithMismatchedTypes ()
       Assert.That (
           () => BusinessObjectProvider.SetProvider (typeof (BindableObjectProviderAttribute), _provider),
           Throws.ArgumentException
-              .With.Message.EqualTo (
-                  "The provider is not compatible with the provider-type required by the businessObjectProviderAttributeType's instantiation."
-                  + "\r\nParameter name: provider"));
+              .With.ArgumentExceptionMessageEqualTo (
+                  "The provider is not compatible with the provider-type required by the businessObjectProviderAttributeType's instantiation.",
+                 "provider"));
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/UnitTests/BusinessObjectProviderTest.cs:159]: whitespace/indent

File: Remotion/Security/UnitTests/DemandPermissionAttributeTest.cs [hunk 9]
```diff
@@ -47,9 +48,9 @@ public void RejectOtherObjectTypes ()
       Assert.That (
           () => new DemandPermissionAttribute (new SimpleType()),
           Throws.ArgumentException
-              .With.Message.EqualTo (
-                  "Item 0 of parameter 'accessTypes' has the type 'Remotion.Security.UnitTests.SampleDomain.SimpleType' instead of 'System.Enum'."
-                  + "\r\nParameter name: accessTypes"));
+              .With.ArgumentExceptionMessageEqualTo (
+                  "Item 0 of parameter 'accessTypes' has the type 'Remotion.Security.UnitTests.SampleDomain.SimpleType' instead of 'System.Enum'.",
+                 "accessTypes"));
```
Comment (MichaelKetting) [inline Remotion/Security/UnitTests/DemandPermissionAttributeTest.cs:53]: whitespace/indent

File: Remotion/Security/UnitTests/Metadata/ClassReflectorTest.cs [hunk 10]
```diff
@@ -140,9 +141,9 @@ public void GetMetadataWithInvalidType ()
       Assert.That (
           () => new ClassReflector().GetMetadata (typeof (Role), _cache),
           Throws.ArgumentException
-              .With.Message.EqualTo (
-                  "Parameter 'type' is a 'Remotion.Security.UnitTests.TestDomain.Role', which cannot be assigned to type 'Remotion.Security.ISecurableObject'."
-                  + "\r\nParameter name: type"));
+              .With.ArgumentExceptionMessageEqualTo (
+                  "Parameter 'type' is a 'Remotion.Security.UnitTests.TestDomain.Role', which cannot be assigned to type 'Remotion.Security.ISecurableObject'.",
+                 "type"));
```
Comment (MichaelKetting) [inline Remotion/Security/UnitTests/Metadata/ClassReflectorTest.cs:146]: whitespace/indent

File: SecurityManager/Core.UnitTests/Domain/Metadata/SecurableClassDefinitionTests/Common.cs [hunk 11]
```diff
@@ -666,9 +667,9 @@ public void GetStatePropertyTest_InvalidName ()
         Assert.That (
             () => orderClass.GetStateProperty ("Invalid"),
             Throws.ArgumentException
-                .With.Message.EqualTo (
-                    "A state property with the name 'Invalid' is not defined for the secureable class definition 'Remotion.SecurityManager.UnitTests.TestDomain.Order'."
-                    + "\r\nParameter name: propertyName"));
+                .With.ArgumentExceptionMessageEqualTo (
+                    "A state property with the name 'Invalid' is not defined for the secureable class definition 'Remotion.SecurityManager.UnitTests.TestDomain.Order'.",
+                   "propertyName"));
```
Comment (MichaelKetting) [inline SecurityManager/Core.UnitTests/Domain/Metadata/SecurableClassDefinitionTests/Common.cs:672]: whitespace/indent

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  public static partial class NUnitExtensions
+  {
+    public static EqualConstraint ArgumentOutOfRangeExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName, int actualValue)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentOutOfRangeException (message: message, paramName: paramName, actualValue: actualValue).Message);
+    }
+
+    public static ContainsConstraint ArgumentExceptionMessageContains (this ConstraintExpression constraintExpression, string message, string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:33]: that one needs to be checked if it's needed.

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public static EqualConstraint ArgumentExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static RegexConstraint ArgumentExceptionMessageMatches (this ConstraintExpression constraintExpression, string message, string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:45]: that one already got canned

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.Matches (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static EndsWithConstraint ArgumentNullExceptionMessageEndsWith (this ConstraintExpression constraintExpression, string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:52]: Probably should change this to ArgumentNullExceptionMessageEqualTo with corresonding semantics. Don't need to pass a message. still should consider, if this is really needed. we have another assertion somewhere where we only go for the parameter name I think? maybe unify towards that variant instead?

File: SharedSource/UnitTests/Development/NUnit/NUnitExtensionsTests.cs [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+namespace Remotion.UnitTests.Development.NUnit
+{
+  [TestFixture]
+  public class NUnitExtensionsTests
+  {
+    [Test]
+    public void AssertThatParameterDoesNotContainWhitespaces_ParameterWithWhitespaces_ThrowsInvalidOperationException ()
+    {
+      var paramName = "param name";
+      Assert.Throws<InvalidOperationException> (() => NUnitExtensions.AssertThatParameterDoesNotContainWhitespaces (paramName));
```
Comment (MichaelKetting) [inline SharedSource/UnitTests/Development/NUnit/NUnitExtensionsTests.cs:30]: Assertion methods private, tests go against the public apis and have to excersise all of them. should wait until we are done kicking out the ones we don't need.

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return new ArgumentOutOfRangeException (message: message, paramName: paramName, actualValue: actualValue).Message;
+    }
+
+    public static void AssertThatMessageContainsWhitespaces (string message)
+    {
+      Assertion.IsTrue (message.Contains (" "), $"The exception message must contain at least one whitespace.\r\nmessage:", message);
+    }
+
+    public static void AssertThatParameterDoesNotContainWhitespaces (string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:77]:
```suggestion
    private static void AssertThatParameterDoesNotContainWhitespaces (string paramName)
```

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public static string CreateArgumentOutOfRangeExceptionMessage (string message, string paramName, object actualValue)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return new ArgumentOutOfRangeException (message: message, paramName: paramName, actualValue: actualValue).Message;
+    }
+
+    public static void AssertThatMessageContainsWhitespaces (string message)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:72]:
```suggestion
    private static void AssertThatMessageContainsWhitespaces (string message)
```

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public static string CreateArgumentExceptionMessage (string message, string paramName)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return new ArgumentNullException (message: message, paramName: paramName).Message;
+    }
+
+    public static string CreateArgumentOutOfRangeExceptionMessage (string message, string paramName, object actualValue)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:65]: Drop

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+      return constraintExpression.With.Message.Matches (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static EndsWithConstraint ArgumentNullExceptionMessageEndsWith (this ConstraintExpression constraintExpression, string paramName)
+    {
+      AssertThatParameterDoesNotContainWhitespaces (paramName);
+      return constraintExpression.With.Message.EndsWith (new ArgumentNullException (paramName: paramName).Message);
+    }
+
+    public static string CreateArgumentExceptionMessage (string message, string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:58]: Drop

File: SharedSource/UnitTests/Utilities/NullableTypeUtilityTest.cs [hunk 20]
```diff
@@ -52,7 +53,7 @@ public void IsNullableType_WithNull_ThrowsArgumentNullException ()
     {
       Assert.That (
           () => NullableTypeUtility.IsNullableType (null!),
-          Throws.TypeOf<ArgumentNullException>().With.Message.EndsWith ("Parameter name: type"));
+          Throws.TypeOf<ArgumentNullException>().With.ArgumentNullExceptionMessageEndsWith ("type"));
```
Comment (MichaelKetting) [inline SharedSource/UnitTests/Utilities/NullableTypeUtilityTest.cs:56]: See comments on ArgNull-method

File: Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs [hunk 21]
```diff
@@ -186,7 +187,7 @@ public void GetValue_InvalidProperty ()
     {
       Assert.That (
           () => _existingDataContainer.GetValue (_nonOrderProperty), 
-          Throws.ArgumentException.With.Message.Contains ("Parameter name: propertyDefinition"));
+          Throws.ArgumentException.With.Message.Contains ("propertyDefinition"));
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs:190]: See ArgNullMessage-check  / helper method

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

File: Remotion/Core/UnitTests/Logging/Log4NetLogManagerTest.cs [hunk 22]
```diff
@@ -185,10 +186,10 @@ public void InitializeConsole_WithSpecificThresholds_InvalidLog4NetLoggerType ()
       logger.Stub (stub => stub.Name).Return ("Foo");
       Assert.That (
           () => _logManager.InitializeConsole (LogLevel.Debug, new LogThreshold (new Log4NetLog (logger), LogLevel.Error)),
-          Throws.ArgumentException.With.Message.Matches (
+          Throws.ArgumentException.With.ArgumentExceptionMessageMatches (
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply Remotion/Core/UnitTests/Logging/Log4NetLogManagerTest.cs:189]: There are 2 tests where the message is checked with `matches`, so it should be no problem to undo the changes and only check the exception message

[the remark this answers is not in the document]
Comment (derlistigelurch) [reply Remotion/Core/UnitTests/Logging/Log4NetLogManagerTest.cs:189]:
```
.With.ArgumentExceptionMessageMatches (
                  "The object to be removed has the same ID as an object in this collection, but is a different object reference.",
                  "domainObject")
```
In this case the `Matches` check could also be an `EqualTo`, because the whole message must be equal and not just a specific pattern.

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs (hunk 12 above)
Comment (derlistigelurch) [reply SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:33]:
```
.With.ArgumentExceptionMessageContains (
                  "The 'PreparingRedirectToSubFunctionState' type only supports WxePermaUrlOptions with the UsePermaUrl-flag set to true.",
                  "parameters"));
```
it's only used once and the check could also be an `EqualTo`, because the whole message and the parameter are checked

File: Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs (hunk 21 above)
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs:190]: What if the message was also validated? This would prevent the parameter from being checked without context.

It would also be possible to replace the `Contains` with  `Matches`, the regex could look something like this `(Parameter) .+(propertyDefinition)`. This would work in both versions and would also prevent the parameter being tested without context

[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs:190]: The regex would mean hardcoding the net45 / net5.0 syntaxes, which we are trying to avoid.

validating the message: possibly, question is, why was this avoided before?

Third option: what does ArgumentException say to passing an empty string as the message? Do we then get only a paramtername-string in the resulting message we could assert with contains?

File: Remotion/Development/UnitTests/Development.UnitTests.csproj [hunk 23]
```diff
@@ -212,6 +212,9 @@
     <Compile Include="..\..\AssemblyInfoShared.cs">
       <Link>Properties\AssemblyInfoShared.cs</Link>
     </Compile>
+    <Compile Include="..\..\..\SharedSource\Development\UnitTesting\NUnit\NUnitExtensions.cs">
+      <Link>SharedSource\UnitTesting\NUnit\NUnitExtensions.cs</Link>
+    </Compile>
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply Remotion/Development/UnitTests/Development.UnitTests.csproj:217]: hmm... this is needed for the tests in `SharedSource`

[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Development/UnitTests/Development.UnitTests.csproj:217]: Ah, because Development.UnitTests links files from SharedSource that have been updated? okay.

File: Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs (hunk 21 above)
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs:190]: > Third option: what does ArgumentException say to passing an empty string as the message? Do we then get only a paramtername-string in the resulting message we could assert with contains?

this `new ArgumentException (message: string.Empty, paramName: "propertyDefinition").Message` returns `\r\nParameter name: propertyDefinition`, i think this the best solution

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs (hunk 14 above)
Comment (derlistigelurch) [reply SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:52]: `ArgumentNullExceptionMessageEndsWith` checks if the parameter name is correct, therefore it can also be replaced with the new method which creates a new message containing only the parameter name string.

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public static EqualConstraint ArgumentExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static ContainsConstraint ArgumentExceptionMessageContains (this ConstraintExpression constraintExpression, string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:40]:
```suggestion
    public static ContainsConstraint ArgumentExceptionMessageWithParamemeterNameEqualTo (this ConstraintExpression constraintExpression, string paramName)
```

File: SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+  public static class NUnitExtensions
+  {
+    public static EqualConstraint ArgumentOutOfRangeExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName, int actualValue)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentOutOfRangeException (message: message, paramName: paramName, actualValue: actualValue).Message);
+    }
+
+    public static EqualConstraint ArgumentExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName)
```
Comment (MichaelKetting) [inline SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs:33]: Please move that one to the top, it's more important than the OutOfRange-check

File: Remotion/Data/DomainObjects.UnitTests/DataManagement/CollectionData/ModificationCheckingDomainObjectCollectionDataDecoratorTest.cs [hunk 26]
```diff
@@ -71,8 +72,10 @@ public void Insert_Duplicate ()
       StubInnerData (_order1);
 
       CheckThrows<ArgumentException> (
-          () => _modificationCheckingDecorator.Insert (0, _order1), 
-          "The collection already contains an object with ID 'Order|5682f032-2f0b-494b-a31c-c97f02b89c36|System.Guid'.\r\nParameter name: domainObject");
+          () => _modificationCheckingDecorator.Insert (0, _order1),
```
[the remark this answers is not in the document]
Comment (raffaelfoidl) [reply Remotion/Data/DomainObjects.UnitTests/DataManagement/CollectionData/ModificationCheckingDomainObjectCollectionDataDecoratorTest.cs:74]: https://github.com/re-motion/Framework/pull/148

File: Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs [hunk 27]
```diff
@@ -1294,9 +1295,9 @@ public void SetDomainObject_InvalidID ()
       Assert.That (
           () => dc.SetDomainObject (domainObject),
           Throws.ArgumentException
-              .With.Message.EqualTo (
-                  "The given DomainObject has another ID than this DataContainer.\r\n"
-                  + "Parameter name: domainObject"));
+              .With.ArgumentExceptionMessageEqualTo (
+                  "The given DomainObject has another ID than this DataContainer.",
+                 "domainObject"));
```
Comment (MichaelKetting) [inline Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerTest.cs:1300]:
```suggestion
                  "domainObject"));
```
I think...

File: SharedSource/UnitTests/Development/NUnit/NUnitExtensionsTests.cs [hunk 28]
```diff
@@ -0,0 +1,114 @@
...
+using NUnit.Framework;
+using Remotion.Development.UnitTesting.NUnit;
+
+namespace Remotion.UnitTests.Development.NUnit
+{
+  [TestFixture]
+  public class NUnitExtensionsTests
+  {
+    [Test]
+    public void ArgumentExceptionMessageEqualTo_MessageAndParameterEqual_Succeeds ()
```
Comment (MichaelKetting) [inline SharedSource/UnitTests/Development/NUnit/NUnitExtensionsTests.cs:28]: Thanks for those tests. What's missing are tests about the two rules (message must contain whitespace. and parmater must not contain whitespace). That's problably gotten lost in the task-handover :)

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

**Review by MichaelKetting — APPROVED** [review]

**Review by MichaelKetting — APPROVED** [review]
